### PR TITLE
Statistics dev tools

### PR DESCRIPTION
**Pull request by bramkragten**
Requires:
- https://github.com/home-assistant/core/pull/56184
- https://github.com/home-assistant/core/pull/55078

## Proposed change

Add statistics dev tools to fix issues with statistics generation.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/developer-tools/statistics/developer-tools-statistics.ts [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            href="https://developers.home-assistant.io/docs/core/entity/sensor"
+            target="_blank"
+          >
+            developer documentation</a
+          >.`,
+      });
+      return;
+    }
+    if (issue.type === "units_changed") {
+      showFixStatisticsUnitsChangedDialog(this, issue);
```
Comment (balloob) [inline src/panels/developer-tools/statistics/developer-tools-statistics.ts:101]: Should we reload this page when it's fixed?

File: src/panels/developer-tools/statistics/developer-tools-statistics.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+  @state() private _data: (StatisticsValidationResult["data"] & {
+    type: string;
+  })[] = [];
+
+  protected firstUpdated() {
+    this._validateStatistics();
+  }
+
+  private _columns: DataTableColumnContainer = {
+    statistic_id: {
```
Comment (balloob) [inline src/panels/developer-tools/statistics/developer-tools-statistics.ts:30]: Would be nice if we detect that it's an entity ID that people can open the more info dialog maybe?

File: src/panels/developer-tools/statistics/developer-tools-statistics.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        .columns=${this._columns}
+        .data=${this._data}
+        noDataText="No issues found!"
+      ></ha-data-table>
+    `;
+  }
+
+  private async _validateStatistics() {
+    const issues = Object.values(await validateStatistics(this.hass));
+    this._data = [];
```
Comment (balloob) [inline src/panels/developer-tools/statistics/developer-tools-statistics.ts:71]: Would it be an idea to show all statistic IDs that we have ?

File: src/panels/developer-tools/statistics/developer-tools-statistics.ts (hunk 2 above)
Comment (balloob) [reply src/panels/developer-tools/statistics/developer-tools-statistics.ts:30]: bonus points if we could open a dialog with a graph with all the available stats data

Comment (balloob) [reply src/panels/developer-tools/statistics/developer-tools-statistics.ts:30]: oh a column with oldest data point maybe?

Comment (balloob) [reply src/panels/developer-tools/statistics/developer-tools-statistics.ts:30]: (ok, future PRs maybe)

File: src/panels/developer-tools/statistics/developer-tools-statistics.ts (hunk 1 above)
Comment (bramkragten) [reply src/panels/developer-tools/statistics/developer-tools-statistics.ts:101]: Oh yeah! That was on my to-do still... Forgot it

File: src/panels/developer-tools/statistics/developer-tools-statistics.ts (hunk 3 above)
Comment (bramkragten) [reply src/panels/developer-tools/statistics/developer-tools-statistics.ts:71]: We could do that... And then sort by issue?

**Review by balloob — DISMISSED** [review]

File: src/data/history.ts [hunk 4]
```diff
@@ -76,6 +76,22 @@ export interface StatisticsMetaData {
   statistic_id: string;
 }
 
+export type StatisticsValidationResult =
+  | StatisticsValidationResultUnspuportedUnit
```
Comment (balloob) [inline src/data/history.ts:80]:
```suggestion
  | StatisticsValidationResultUnsupportedUnit
```

**Review by balloob — DISMISSED** [review]

**Pull request merged**